modules/kargoBidAdapter.js

```diff
@@ -4,286 +4,484 @@ import { registerBidder } from '../src/adapters/bidderFactory.js';
+  }
+}
 
-  _getCrb() {
-    let localStorageCrb = spec._getCrbFromLocalStorage();
-    if (Object.keys(localStorageCrb).length) {
-      return localStorageCrb;
-    }
-    return spec._getCrbFromCookie();
-  },
-
-  _getLocalStorageSafely(key) {
-    try {
-      return storage.getDataFromLocalStorage(key);
-    } catch (e) {
-      return null;
-    }
-  },
-
-  _getUserIds(tdid, usp, gdpr) {
-    const crb = spec._getCrb();
-    const userIds = {
-      kargoID: crb.lexId,
-      clientID: crb.clientId,
-      crbIDs: crb.syncIds || {},
-      optOut: crb.optOut,
-      usp: usp
-    };
 
-    try {
-      if (gdpr) {
-        userIds['gdpr'] = {
-          consent: gdpr.consentString || '',
-          applies: !!gdpr.gdprApplies,
-        }
+
+function getLocalStorageSafely(key) {
+  try {
+    return STORAGE.getDataFromLocalStorage(key);
+  } catch (e) {
+    return null;
+  }
+}
+
+function getUserIds(tdid, usp, gdpr, eids) {
+  const crb = spec._getCrb();
+  const userIds = {
+    crbIDs: crb.syncIds || {}
+  };
+
+  if (tdid) {
+    userIds.tdID = tdid;
+  }
+
+  if(usp) {
+    userIds.usp = usp
+  }
+
+  try {
+    if (gdpr) {
+      userIds['gdpr'] = {
+        consent: gdpr.consentString || '',
+        applies: !!gdpr.gdprApplies,
       }
-    } catch (e) {
     }
-    if (tdid) {
-      userIds.tdID = tdid;
+  } catch (e) {
+  }
+  
+  if(crb.lexId != null) {
+    userIds.kargoID = crb.lexId
+  }
+
+  if(crb.clientId !=  null) {
+    userIds.clientID = crb.clientId
+  }
+
+  if(crb.optOut != null) {
+    userIds.optOut = crb.optOut
+  }
+
+  if(eids != null) {
+    userIds.eids = eids
+  }
+
+  return userIds;
+}
+
+function getClientId() {
+  const crb = spec._getCrb();
+  return crb.clientId;
+}
+
+function getAllMetadata(bidderRequest) {
+  return {
+    pageURL: bidderRequest?.refererInfo?.page,
+    rawCRB: STORAGE.getCookie(CERBERUS.KEY),
+    rawCRBLocalStorage: getLocalStorageSafely(CERBERUS.KEY)
+  };
+}
+
+function getRequestCount() {
+  if (lastPageUrl === window.location.pathname) {
+    return ++requestCounter;
+  }
+  lastPageUrl = window.location.pathname;
+  return requestCounter = 0;
+}
+
+function sendTimeoutData(auctionId, auctionTimeout) {
+  let params = {
+    aid: auctionId,
+    ato: auctionTimeout,
+  };
+
+  try {
+    let timeoutRequestUrl = buildUrl({
+      protocol: 'https',
+      hostname: 'krk.kargo.com',
+      pathname: '/api/v1/event/timeout',
+      search: params
+    });
+
+    triggerPixel(timeoutRequestUrl);
+  } catch (e) {}
+}
+
+function getImpression(bid) {
+  const imp = {
+    id: bid.bidId,
+    tid: bid.transactionId,
+    pid: bid.params.placementId,
+    code: bid.adUnitCode
+  }
+
+  if (bid.floorData != null && bid.floorData.floorMin > 0) {
+    imp.floor = bid.floorData.floorMin
+  }
+
+  if (bid.bidRequestsCount > 0) {
+    imp.bidRequestCount = bid.bidRequestsCount
+  }
+
+  if (bid.bidderRequestsCount > 0) {
+    imp.bidderRequestCount = bid.bidderRequestsCount
+  }
```

Comment:
i think can get rid of one of them?

Reply:
These are different fields. The naming is confusing but they are different